awx_collection/plugins/modules/tower_workflow_launch.py

```diff
@@ -70,97 +95,118 @@
 
 - name: Launch a Workflow with parameters without waiting
   tower_workflow_launch:
-    workflow_template: "Test workflow"
+    name: "Test workflow"
     extra_vars: "---\nmy: var"
```

Comment:
These `extra_vars` seem... questionable.

In every other case, we have changed the type of extra_vars from a string to a dict. It's only going to create user confusion to leave one as the old string type. Being a dict doesn't mean it _can't_ take a string, it's just more limited. This very generous string-to-dict coercion is exactly what we've deprecated in all other places (and should do here too).

We should also add `extra_vars` to one of the tasks in the integration tests. Piggy-backing it onto an existing task would be 👍 with me.

changing the type may still requiring doing `json.loads` to satisfy the API, don't know, just have to write test and make it work.